working/augmentation-libraries/feature-specification.md

```diff
@@ -545,13 +549,25 @@ More specifically:
     with the body of the augmenting getter. Inside the augmenting getter’s
     body, an `augmented` expression executes the augmented getter’s body.
 
+    Augmenting getter declarations may also omit the body, in order to only
+    augment the metadata or doc comments of the getter. In this case the body of
+    the augmented getter is not altered.
+
+    Only concrete getters are allowed to augment with metadata or doc comments.
```

Comment:
I don't understand what this sentence means. Is it saying that an augmenting getter with a `;` body *is* concrete (i.e. not abstract) or something else?

Reply:
I meant to say only non-synthetic getters can be augmented, but concrete is not the correct term as it implies having a body etc.

Updated the wording to say that synthetic getters (and setters below) cannot be augmented with metadata/doc comments.

We could possibly just remove this restriction, but I don't really know what the behavior should be if we did allow it.

Reply:
Heh, I also don't know what a "synthetic" getter or setter is. 🙃

Reply:
The ones introduced by variable declarations.

Reply:
Maybe it is just an analyzer term, but basically they are any declarations which are not specified by the user explicitly, and are implicitly added based on some other declaration.

Not sure if the default constructor is considered synthetic or not.

Reply:
Ah, I see. In that case, I don't see the value of this restriction. You can augment a variable declaration with a getter. It seems weird that this is OK:

```dart
int x;

augment int get x {
  print('wrapped!');
  return augmented;
}
```

But then this becomes an error:

```dart
int x;

/// Comment.
augment int get x {
  print('wrapped!');
  return augmented;
}
```

I can't think of any other places where adding a comment yields an error.

Can we just remove this restriction?

Reply:
Agree. We can say that doc-comments on augmenting declarations for implicit getters/setters will (or may) be ignored by tooling.

Annotations on the other hand, they're harder to ignore and it makes sense to not allow them.
(As long as you can declare the base variable without adding an initializer, just to update the docs.)

Reply:
Yes the restriction is primarily because of annotations, especially macro annotations. We could allow doc comments though.